djvm/src/main/kotlin/net/corda/djvm/rules/implementation/DisallowNonDeterministicMethods.kt

```diff
@@ -15,36 +16,94 @@ import sandbox.net.corda.djvm.rules.RuleViolationError
 object DisallowNonDeterministicMethods : Emitter {
 
     private const val CHARSET_PACKAGE = "sun/nio/cs/"
+    private const val SUN_SECURITY_PROVIDERS = "sun/security/jca/Provider"
     private val MONITOR_METHODS = setOf("notify", "notifyAll", "wait")
-    private val CLASSLOADING_METHODS = setOf("defineClass", "loadClass", "findClass")
+    private val CLASSLOADING_METHODS = setOf("defineClass", "findClass")
```

Comment:
Can `defineClass` be invoked directly by any class in the sandbox? That seems strange, as it would jump into native code.

Reply:
Probably not, but why take chances?